src/dotnet/APIView/APIView/TypeReference.cs

```diff
@@ -19,56 +19,19 @@ public TypeReference()
         public TypeReference(Token[] tokens)
         {
             this.Tokens = tokens;
-            this.IsString = this.Tokens.Last().IsString;
-            this.Type = this.Tokens.Last().Type;
-        }
-
-        public TypeReference(AttributeData data)
-        {
-            var tokens = new List<Token>();
-            foreach (var part in data.AttributeClass.ToDisplayParts())
-            {
-                tokens.Add(new Token(part));
-            }
-            this.Tokens = tokens.ToArray();
-            this.Type = this.Tokens.Last().Type;
-            this.IsString = this.Tokens.Last().IsString;
-        }
-
-        public TypeReference(IEventSymbol symbol)
-        {
-            var tokens = new List<Token>();
-            foreach (var part in symbol.Type.ToDisplayParts())
-            {
-                tokens.Add(new Token(part));
-            }
-            this.Tokens = tokens.ToArray();
-            this.Type = this.Tokens.Last().Type;
-            this.IsString = this.Tokens.Last().IsString;
-        }
-
-        public TypeReference(IFieldSymbol symbol)
-        {
-            var tokens = new List<Token>();
-            foreach (var part in symbol.Type.ToDisplayParts())
-            {
-                tokens.Add(new Token(part));
-            }
-            this.Tokens = tokens.ToArray();
+            this.IsString = false;
             this.Type = this.Tokens.Last().Type;
-            this.IsString = this.Tokens.Last().IsString;
         }
 
-        public TypeReference(IMethodSymbol symbol)
+        public TypeReference(ISymbol symbol)
         {
             var tokens = new List<Token>();
-            foreach (var part in symbol.ReturnType.ToDisplayParts())
+            foreach (var part in symbol.ToDisplayParts())
             {
                 tokens.Add(new Token(part));
             }
             this.Tokens = tokens.ToArray();
             this.Type = this.Tokens.Last().Type;
-            this.IsString = this.Tokens.Last().IsString;
         }
 
         public TypeReference(INamedTypeSymbol symbol)
```

Comment:
Do you really need this one? Can enum related logic be in TypeAPIV?

Reply:
What do you mean?

Reply:
```
if (symbol.EnumUnderlyingType != null)
            {
                foreach (var part in symbol.EnumUnderlyingType.ToDisplayParts())
                {
                    tokens.Add(new Token(part));
                }
            }
```
this part, it's not really correct, instead of getting tokens for enum name you are getting them for underlying type which is int most of the time.